src/Logging.fs

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 Cognite AS
+
+namespace Oryx
+
+open System
+open System.Collections.Generic
+open System.Net.Http
+open System.Text.RegularExpressions
+open Microsoft.Extensions.Logging
+
+[<AutoOpen>]
+module Logging =
+
+    let setLogger (logger: ILogger) (next: NextFunc<HttpResponseMessage,'r, 'err>) (context: HttpContext) =
+        next { context with Request = { context.Request with Logger = Some logger } }
+
+    let setLogLevel (logLevel: LogLevel) (next: NextFunc<HttpResponseMessage,'r, 'err>) (context: HttpContext) =
+        next { context with Request = { context.Request with LogLevel = logLevel } }
+
+    // Pre-compiled
+    let private reqex = Regex(@"\{(.+?)\}", RegexOptions.Multiline)
```

Comment:
I think you actually have to specify option `RegexOptions.Compiled` to precompile it.

Reply:
Also, I don't know if the format strings are somehow standard, but Serilog certainly allows to specify some formatting directives after a colon (for example `{Property:options}`).

Reply:
Checked with `Microsoft.Extensions.Logging` and it will let you specify and use formatting directives such as `{Property:options}`.

Reply:
But even Serilog specifies it must be a valid .NET identifier: https://github.com/serilog/serilog/wiki/Writing-Log-Events